static/js/popovers.js

```diff
@@ -559,6 +560,14 @@ export function toggle_actions_popover(element, id) {
             view_source_menu_item = $t({defaultMessage: "View message source"});
         }
 
+        // For should_display_mark_as_unread, we want to only offer
+        // the ability to mark messages unread in streams that the
+        // user is subscribed to. This does not apply to messages
+        // that are not stream messages.
```

Comment:
Let's replace this comment with an explanation of why this check is important/correct -- that marking messages as unread in such a stream would violate our invariant that all unread messages are accessible to the user in the left sidebar.

One can generally figure out what the code is doing by reading it -- so comments summarizing what it does are relatively low value (they can help when the code itself is hard to read; but this code is fairly readable just from the variable names and function calls). But the really invaluable comments are the ones that explain why we're doing something, especially when, like this, it's a subtle interaction with considerations that are not at all local to this function.